server/db/connection.js

```diff
@@ -15,13 +18,88 @@ class Db {
   connectDb() {
     let dbOptions = {
       dbName: "" + dbName,
-    }
-    mongoose.connect(uri, dbOptions)
+    };
+    mongoose.connect(uri, dbOptions);
+    console.log("Connected to MongoDB");
   }
 
   disconnectDb() {
-    mongoose.connection.close()
+    mongoose.connection.close();
+  }
+
+  async getAll(collectionSchema) {
+    const collection = collectionSchema.find();
+    return collection;
+  }
+
+  // CRUD Functions
+  async addUser(userFname, userLname, userRole, userId, userPw) {
+    const user = await User.create({
+      fname: userFname,
+      lname: userLname,
+      role: userRole,
+      user_id: userId,
+      pw: userPw,
+    });
+    return user;
+  }
+
+  async addCourse(courseId, courseNumber, courseDept, courseProfId) {
+    const course = await Course.create({
+      course_id: courseId,
+      number: courseNumber,
+      dept: courseDept,
+      prof_id: courseProfId,
+    });
+  }
+
+  async addSection(sectionStudentIds, sectionCourseId, sectionName) {
+    const section = await Section.create({
+      student_ids: sectionStudentIds,
+      course_id: sectionCourseId,
+      name: sectionName,
+    });
+    return section;
+  }
+
+  async getUser(userId) {
+    const user = await User.findOne({ user_id: userId });
```

Comment:
This comment goes for all the functions you defined.

I may be wrong and mongoose might already be handling this by simply returning null/undefined but don't you need error handling in case the mongoose function call doesn't work (i.e. the promise is rejected?)

Reply:
https://stackoverflow.com/a/76826713

Mongoose does not handle errors and will just throw the errors up. We can decide next sprint if the API level or the DB object should handle the error